src/generator/state-machine-to-intermediate.js

```diff
@@ -154,18 +154,37 @@ const getIntermediateFormOf = ({left, right, operator, out}, isSecondarySignal)
     }
 };
 
+// Helper function to generate next available numeric label
+function generateNextLabel(states) {
```

Comment:
```suggestion
function endOfLastNumericState(states) {
    const ends = states
        .filter(({state}) => typeof state === 'number')
        .map(({statements}) => statements.map(s => s.start + s.operations.length))
        .flat();
    return Math.max(...ends) + 1;
```

I don't think `.flat` was standard when this code was first written, but we can use it now.

Reply:
Seems good, I just fixed a bug when ends is empty but this works